tests/integration/ambient/baseline_test.go

```diff
@@ -3584,3 +3584,156 @@ spec:
+			ztunnelLabelPatch := fmt.Sprintf(`{"metadata":{"labels": {"%s": "http"}}}`, "networking.istio.io/tunnel")
+			_, err = k8sPods.Patch(context.Background(), ztunnelPod.Name, types.StrategicMergePatchType, []byte(ztunnelLabelPatch), patchOpts)
+			if err != nil {
+				tc.Fatalf("Failed to label ztunnel pod %s: %v", ztunnelPod.Name, err)
+			}
+
+			tc.Cleanup(func() {
+				// Cleanup the label
+				cleanupPatch := fmt.Sprintf(`{"metadata":{"labels": {"%s": null}}}`, "networking.istio.io/tunnel")
+				_, err = k8sPods.Patch(context.Background(), ztunnelPod.Name, types.StrategicMergePatchType, []byte(cleanupPatch), patchOpts)
+				if err != nil {
+					tc.Logf("Failed to remove networking.istio.io/tunnel label from ztunnel pod %s: %v", ztunnelPod.Name, err)
+				}
+				tc.Logf("Removed networking.istio.io/tunnel label from ztunnel pod %s", ztunnelPod.Name)
+			})
+
+			tc.Logf("Using client %s (%s) to query ztunnel %s (%s) metrics on port %d. Expecting transparent HBONE.",
+				clientInstance.Config().Service, clientInstance.WorkloadsOrFail(tc)[0].PodName(), ztunnelPod.Name, ztunnelPodIP, ztunnelMetricsPort)
+
+			// Client calls ztunnel's `/metrics` endpoint.
+			// This request should be intercepted by clientInstance's ztunnel,
+			// and an HBONE connection made to the target ztunnel's inbound (15008),
+			// which then proxies to its internal metrics server (15020).
+			opts := echo.CallOptions{
+				Address: ztunnelPodIP,
+				Port:    echo.Port{ServicePort: ztunnelMetricsPort, Name: "http-ztunnel-metrics", Protocol: protocol.HTTP},
+				Scheme:  scheme.HTTP,
+				HTTP:    echo.HTTP{Path: "/metrics"},
+				Check:   check.And(check.OK(), check.BodyContains("# TYPE")), // Check for Prometheus format
+			}
+			clientInstance.CallOrFail(tc, opts)
+			tc.Logf("Successfully called ztunnel /metrics endpoint via HTTP from %s", clientInstance.WorkloadsOrFail(tc)[0].PodName())
+
+			// Verify Prometheus L4 telemetry for the HBONE connection to ztunnel
+			// The ztunnel pod itself is the destination workload for this specific HBONE connection.
+			// sourceWorkloadPodName := clientInstance.WorkloadsOrFail(tc)[0].PodName() // For istio_tcp_connections_opened_total, source_workload is pod name
+			sourceNamespace := clientInstance.Config().Namespace.Name()
+			sourceSA := clientInstance.Config().AccountName()
+			sourceWorkloadLabel := clientInstance.Config().Service + "-" + clientInstance.Config().Version
+
+			query := prometheus.Query{
+				Metric: "istio_tcp_connections_opened_total",
+				Labels: map[string]string{
+					"reporter":                       "destination",
+					"connection_security_policy":     "mutual_tls",
+					"destination_workload_namespace": istioSystemNS,
+					"destination_workload":           "ztunnel",
+					"destination_principal":          fmt.Sprintf("spiffe://%s/ns/%s/sa/%s", trustDomain, istioSystemNS, ztunnelServiceAccount), // Restored usage of ztunnelServiceAccount
```

Comment:
Lint doesn't like this long line. I don't think the comment is relevant and could just be removed